src/python/pants/BUILD

```diff
@@ -55,3 +55,39 @@ pex_binary(
 # use `entry_points` to link to the actual native code, so clang/gcc do not need to build any
 # native code. This is just a dummy file.
 resources(name="resources", sources=["py.typed", "dummy.c"])
+
+PANTS_ARTIFACT_DEPENDENCIES = (
+    "3rdparty/python#ansicolors",
+    "3rdparty/python#chevron",
+    "3rdparty/python#fasteners",
+    "3rdparty/python#ijson",
+    "3rdparty/python#node-semver",
+    "3rdparty/python#packaging",
+    "3rdparty/python#pex",
+    "3rdparty/python#psutil",
+    "3rdparty/python#python-lsp-jsonrpc",
+    "3rdparty/python#PyYAML",
+    "3rdparty/python#setproctitle",
+    "3rdparty/python#setuptools",
+    "3rdparty/python#toml",
+    "3rdparty/python#types-PyYAML",
+    "3rdparty/python#types-setuptools",
+    "3rdparty/python#types-toml",
+    "3rdparty/python#typing-extensions",
+)
+
+
+__dependencies_rules__(
+    (
+        # source modules can only depend on a restricted subset of 3rd-party requirements that
+        # published Pants artifact will require; test modules do not have any restrictions and
+        # can depend on any 3rd party requirement as they aren't included in the final artifact
+        (
+            {"type": python_sources},
+            PANTS_ARTIFACT_DEPENDENCIES,
```

Comment:
@kaos this is where having https://github.com/pantsbuild/pants/issues/19507 implemented would come in handy, I think. A tuple is not terrible, but can't be accessed outside of this BUILD file, right?

Reply:
> @kaos this is where having #19507 implemented would come in handy, I think. A tuple is not terrible, but it can't be accessed outside of this BUILD file, right?

Will reply to this first, before having looked at the rest.

The tuple could be put in a macro, as such it would be globally available.